common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -331,6 +363,17 @@ bool base_wlan_hal_dwpal::fsm_setup()
 
             m_fd_ext_events = -1;
 
+            // detach nl socket from main vap
+            if (!m_dwpal_nl_ctx) {
```

Comment:
You should check if the context IS valid:
`if (m_dwpal_nl_ctx) { ... }`

Reply:
fixed